### DISCUSSION TITLE
Welcome to NCEPLIBS Discussions!

### DISCUSSION THREAD

**Discussion by arunchawla-NOAA** [NOAA-EMC/NCEPLIBS#195] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by arunchawla-NOAA** [comment]
Hello

This Discussion thread has been started to identify features that we would like to see in NCEPLIBS. Once a path forward is identified we can then open relevant issues in corresponding repos

**Comment by aerorahul** [comment]
It would be great if the libraries (e.g. bufr) are properly overloaded.  i.e. there is no compile time choices for single, double or mixed precision.

**Reply by kgerheiser** [reply]
That would simplify our build system considerably instead of building 3 versions of each library, as well as the type-checking benefits. 

It would also make the code more portable because you won't have to hardcode the flags for each compiler.

In some numerical libraries (ip, sp) where you just need to support different kinds of reals it might be advantageous to have the code work on `real64` and overload the interface by simply casting `real32` to `real64`. Though, there could be memory implications.

In other cases you have to resort to some hack using the pre-processor or Fortran `include()`.

**Comment by edwardhartnett** [comment]
Indeed yes, that will be great. See https://github.com/NOAA-EMC/NCEPLIBS-bufr/issues/78.
